### ISSUE TITLE
opam package for OCaml 4.02?

### ISSUE THREAD

**Issue by vogler** [cil-project/cil#18]
It would be nice to have an opam package for OCaml >= 4.02, since the current one leads to segmentation faults (and cilly doesn't terminate).
Also, for the develop-branch, I had to add `package(cil.all-features)` to my `_tags` file, which isn't available using the opam package. Could we make that compatible somehow?

**Comment by charlesxiong** [comment]
Yeah,I have the same question about compiling goblint . Can anyone solve this ?

**Comment by vogler** [comment]
Concerning Goblint: You should be able to run `make setup` which pins cil to the develop branch. You need at least opam 1.2.0 for this to work. If you still have problems, open an issue there.

**Comment by charlesxiong** [comment]
If I run `make setup` to compile goblint(using `opam pin -y add cil https://github.com/kerneis/cil.git#develop`) , it will produce an error 
How do I modify the cil version ?

```
[cil] Fetching https://github.com/kerneis/cil/tree/develop
# opam-version 1.2.0
# os           linux
# command      git fetch origin +HEAD:refs/remotes/opam-ref
# path         /home/charles/.opam/4.02.1/packages.dev/cil
# exit-code    128
# env-file     /home/charles/.opam/log/log-28148-6d07ae.env
# stdout-file  /home/charles/.opam/log/log-28148-6d07ae.out
# stderr-file  /home/charles/.opam/log/log-28148-6d07ae.err
### stderr ###
# fatal: repository 'https://github.com/kerneis/cil/tree/develop/' not found
```

**Comment by vogler** [comment]
Haha, interesting.
It should say `[cil] Fetching https://github.com/kerneis/cil.git#develop`.
What shell is your `/bin/sh`? Apparently bash has no problem with the `#`.
Anyhow, I [fixed](https://github.com/goblint/analyzer/commit/0f1b1399545bb4d548abafcc34d336447fb6c5f8) `make.sh` by quoting the URL.

**Comment by charlesxiong** [comment]
Still the same question if I quote the URL .  My machine is Ubuntu 14.10 , bash shell . 
Actually If I use `opam pin -y add cil "https://github.com/kerneis/cil#develop"` .The url still redirect to the `[cil] Fetching https://github.com/kerneis/cil/tree/develop`  and produce an error .

**Comment by kerneis** [comment]
A release is hopefully coming, but in the meantime, develop is the 
default branch on github anyway, so you should be able to just give it 
github.com/kerneis/cil. Or did I miss something?

**Comment by vogler** [comment]
You're right, it also works without specifying the branch.
I don't know why it doesn't work in your case, but without the branch it should be fine:

```
$ git clone https://github.com/kerneis/cil/tree/develop
Cloning into 'develop'...
fatal: repository 'https://github.com/kerneis/cil/tree/develop/' not found

$ git clone https://github.com/kerneis/cil/
Cloning into 'cil'...
```

**Comment by charlesxiong** [comment]
Yeah , knowing the small detail that cil default branch is develop ,I compiled the goblint and it worked    .  Thank you all for answering my questions  - vogler and kerneis . ^-^
